web-client/src/views/AppComponent.tsx

```diff
@@ -302,6 +303,7 @@ export const AppComponent = connect(
         {showModal === 'AppMaintenanceModal' && <AppMaintenanceModal />}
         {showModal === 'GenericErrorModal' && <GenericErrorModal />}
         {showModal === 'FileUploadErrorModal' && <FileUploadErrorModal />}
+        {showModal === 'VerifyNewEmailModal' && <VerifyNewEmailModal />}
```

Comment:
Since the banner can be shown on any page, the modal needs to be available across the app. I removed it from the place it was appearing "locally" before and added it here for "global" display.